src/main/java/com/google/codeu/servlets/ChartServlet.java

```diff
@@ -0,0 +1,39 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Message;
+import com.google.gson.Gson;
+
+/**
+ * Handles fetching all messages for the public feed.
+ */
+@WebServlet("/messagechart")
+public class ChartServlet extends HttpServlet{
+
+    private Datastore datastore;
+
+    @Override
+    public void init() {
+        datastore = new Datastore();
+    }
+
+    /**
+     * Gets messages from datastore and prints them.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        List<Message> msgList = datastore.getMessages(null);
```

Comment:
Better to call `datastore.getAllMessages()` instead. It's clearer (and you also don't need to change `getMessages` to support null).